.github/workflows/github-actions-on-label-create.yml

```diff
@@ -1,33 +1,69 @@
 name: Labelled Ready to Sync Public
+
 on:
   pull_request:
     types: [labeled]
-  # Allows you to run this workflow manually from the Actions tab
-  workflow_dispatch:
+
+env:
+  PRIVATE_OWNER: ${{ github.event.pull_request.head.repo.owner.login }}
+  PRIVATE_REPO: ${{ github.event.pull_request.head.repo.name }}
+  PRIVATE_BRANCH: ${{ github.event.pull_request.head.ref }}
+  # Staging repository
+  STAGING_OWNER: The-OpenROAD-Project-staging
+  STAGING_REPO: OpenROAD
+  STAGING_BRANCH: ${{ github.event.pull_request.head.ref }}
+  # Upstream repository
+  UPSTREAM_OWNER: The-OpenROAD-Project
+  #UPSTREAM_REPO: ${{ github.event.pull_request.base.repo.name }}
+  UPSTREAM_BRANCH: master
 
 jobs:
-  label:
+  Push-To-Staging:
+    name: Push to staging
     runs-on: ubuntu-latest
-    if: github.repository_owner == 'The-OpenROAD-Project-private' && github.event.label.name == 'Ready To Sync Public'
+
+    permissions:
+      # Read-only access so we don't accidentally try to push to *this* repository.
+      contents: read
+      # Pull request write access, so we can remove the label.
+      issues: write
+      pull-requests: write
+      # Deployment write access, so we can add the nice deployment info.
+      deployments: write
+
+    # Only run on the private repository.
+    if: github.event.repository.private && github.event.label.name == 'Ready To Sync Public'
     steps:
-      - name: Check out repository code
-        uses: actions/checkout@v2
-        with:
-            ref: ${{github.event.pull_request.head.ref}}
-            fetch-depth: 0
-            token: ${{ secrets.GITHUBPUBLICREPOTOKEN }}
-      
-      - name: run security scan
-        uses: The-OpenROAD-Project/actions/security_scan_on_push@main
-        
-      - name: change remote to staging
-        run: git remote set-url origin https://github.com/The-OpenROAD-Project-staging/OpenROAD.git
-      - name: git status
-        run: git status
-      - name: Current label
-        run: echo ${{github.event.label.name}}
-      - name: pr info
-        run: echo ${{toJson(github.event.pull_request.head.ref)}} 
-      - name: push to staging
-        run: git push
-        
+    - name: Removing label '${{ github.event.label.name }}'
+      uses: The-OpenROAD-Project/actions/remove_label@main
```

Comment:
I don't understand https://github.com/The-OpenROAD-Project/actions/blob/main/remove_label/action.py#L50.
What happens if the PR is filed without being connected to an issue?

Reply:
Fundamentally, PRs **are** Issues in GitHub. Even so, this action only triggers when someone puts the correct label on a pull request.